auth/src/test/java/ApiKeySnippetsIT.java

```diff
@@ -0,0 +1,142 @@
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ApiKeySnippetsIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String CREDENTIALS = System.getenv("GOOGLE_APPLICATION_CREDENTIALS");
+  private static Key API_KEY;
+  private ByteArrayOutputStream stdOut;
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setup()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+
+    API_KEY = CreateApiKey.createApiKey(PROJECT_ID);
+
+    stdOut.close();
+    System.setOut(out);
+  }
+
+  @AfterClass
+  public static void cleanup()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+
+    String apiKeyId = getApiKeyId(API_KEY);
+    DeleteApiKey.deleteApiKey(PROJECT_ID, apiKeyId);
+    assertThat(stdOut.toString()).contains(
+        String.format("Successfully deleted the API key: %s", API_KEY.getName()));
+
+    stdOut.close();
+    System.setOut(out);
+  }
+
+  private static String getApiKeyId(Key apiKey) {
+    return apiKey.getName().split("/")[5];
+  }
+
+  @Before
+  public void beforeEach() {
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @After
+  public void afterEach() {
+    stdOut = null;
+    System.setOut(null);
+  }
+
+  @Test
+  public void testLookupApiKey() throws IOException {
+    LookupApiKey.lookupApiKey(API_KEY.getKeyString());
+    assertThat(stdOut.toString()).contains(
+        String.format("Successfully retrieved the API key name: %s", API_KEY.getName()));
+  }
+
+  @Test
+  public void testRestrictApiKeyAndroid()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    RestrictApiKeyAndroid.restrictApiKeyAndroid(PROJECT_ID, getApiKeyId(API_KEY));
+    assertThat(stdOut.toString()).contains(
+        String.format("Successfully updated the API key: %s", API_KEY.getName()));
+  }
+
+  @Test
+  public void testRestrictApiKeyApi()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    RestrictApiKeyApi.restrictApiKeyApi(PROJECT_ID, getApiKeyId(API_KEY));
+    assertThat(stdOut.toString()).contains(
+        String.format("Successfully updated the API key: %s", API_KEY.getName()));
```

Comment:
nit: The wider Google Java style guide says that you should prefer using a variable to avoid line-wrapping, or if line-wrapping is necessary. 

For example, you could do the following here:
```suggestion
    String goal = String.format("Successfully updated the API key: %s", API_KEY.getName());
    assertThat(stdOut.toString()).contains(goal);
```

Reply:
Committed the suggestion in all places. Thanks.